### PR TITLE
Add dataset tests

### PR DESCRIPTION
**Pull request by sphamba**
# Prework

- [x] I understand and agree to this repository's [code of conduct](https://github.com/ghiggi/gpm_api/blob/main/CODE_OF_CONDUCT.md).
- [x] I understand and agree to this repository's [contributing guidelines](https://github.com/ghiggi/gpm_api/blob/main/CONTRIBUTING.md).
- [x] I have already submitted an [issue](https://github.com/ghiggi/gpm_api/issues) or [discussion thread](https://github.com/ghiggi/gpm_api/discussions) to discuss my idea with the maintainers.

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Documentation
- [ ] Tutorial
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and communicate accordingly:

**The PR fulfills these requirements:**

- [ ] It's submitted to the branch named as follow :
  - Fix a bug: `bugfix-<some_key>-<word>`
  - Improve the doc: `doc-<some_key>-<word>`
  - Improve a tutorial `tutorial-<some_key>-<word>`
  - Add a new feature: `feature-<some_key>-<word>`
  - Refactor some code: `refactor-<some_key>-<word>`
  - Optimize some code: `optimize-<some_key>-<word>`
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] Don't forget to link PR to issue if you are solving one.
- [ ] All tests are passing.
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

# Summary

- add script to generate small hdf5 files from downloaded raw files, and netcdf files from `open_granule`.
  - [ ] does not seem to work with all raw files (eg IMERG)
  - [ ] relies on `start_date` to be added in `product_def.yml` for all products
- add tests for granule
  - [ ] test comparing small hdf5 files with saved netcdf file fails
- add tests for groups and variables

### CODE REVIEW

File: gpm_api/tests/dataset/generate_test_dataset_assets.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+
+from dateutil.relativedelta import relativedelta
+
+from gpm_api.configs import get_gpm_username, get_gpm_password
+from gpm_api.dataset.granule import open_granule
+from gpm_api.io import download, products
+from gpm_api.io.pps import find_pps_filepaths
+
+
+# Create asset directories #####################################################
```
Comment (ghiggi) [inline gpm_api/tests/dataset/generate_test_dataset_assets.py:13]: Maybe we could find a better name than asset.
`generate_test_dataset_granules.py` !

File: gpm_api/tests/dataset/generate_test_dataset_assets.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    src_file.close()
+    dst_file.close()
+
+
+def cut_raw_assets():
+    filenames = os.listdir(raw_assets_dir_path)
+
+    for filename in filenames:
+        print(f"Cutting {filename}")
+        raw_asset_filepath = os.path.join(raw_assets_dir_path, filename)
```
Comment (ghiggi) [inline gpm_api/tests/dataset/generate_test_dataset_assets.py:231]: We should remove the raw granules and just keep the cutted ones in the GitHub repo !
Let's discuss on monday on the structure of the test_dataset data directory

**Review by ghiggi — APPROVED** [review]
LGTM.

File: gpm_api/tests/dataset/generate_test_dataset_assets.py (hunk 2 above)
Comment (sphamba) [reply gpm_api/tests/dataset/generate_test_dataset_assets.py:231]: Yes that is what I thought!

**Pull request merged**